When LDFLAGS, CXXFLAGS and others is defined in system environment, the value defined
in system environment should be appended to env, rather than replace.

--- a/SConstruct
+++ b/SConstruct
@@ -3,7 +3,7 @@
 
 locales = []
 
-cflags = '-O2 -g -pipe'
+cflags = '-O2 -g -pipe '
 version = '2.0.3'
 
 AddOption('--prefix', dest='prefix', metavar='DIR',
@@ -27,7 +27,7 @@
     for (x, y) in envvar:
         if x in os.environ:
             print 'Warning: you\'ve set %s in the environmental variable!' % x
-            env[y] = os.environ[x]
+            env[y] += os.environ[x]
 
 env = Environment(ENV=os.environ,
                   CFLAGS=cflags, CXXFLAGS=cflags, 
